### PR TITLE
Allow removal of subtypes from submission

### PR DESCRIPTION
**Pull request by trslater**

### CODE REVIEW

File: alcs-frontend/src/app/features/notice-of-intent/proposal/preparation/preparation.component.ts [hunk 1]
```diff
@@ -65,69 +65,22 @@ export class PreparationComponent implements OnInit {
...
-    }
-
-    return false;
-  }
-
-  async saveSubtypes($event: string | string[] | null) {
-    if (this._noticeOfIntent && $event instanceof Array) {
-      const selectedCodes = $event.map((label) => this.subtypes.find((subtype) => subtype.label === label)!.code);
+  async saveSubtypes(selectedLabels: string | string[] | null) {
+    console.log(selectedLabels);
```
Comment (mhuseinov) [inline alcs-frontend/src/app/features/notice-of-intent/proposal/preparation/preparation.component.ts:78]: looks like a debug leftover

File: alcs-frontend/src/app/features/notice-of-intent/proposal/preparation/preparation.component.ts [hunk 2]
```diff
@@ -59,75 +59,24 @@ export class PreparationComponent implements OnInit {
...
       this.selectableTypes = subtypes.map((subtype) => {
-        let disabled = false;
-
-        disabled = this.isSubtypeDisabled(submission, subtype);
-
         return {
           label: subtype.label,
           value: subtype.label,
-          disabled,
+          disabled: false,
```
Comment (mhuseinov) [inline alcs-frontend/src/app/features/notice-of-intent/proposal/preparation/preparation.component.ts:69]: I think disabled is optional parameter so you could simply ignore it here and it should act as false

Comment (trslater) [reply alcs-frontend/src/app/features/notice-of-intent/proposal/preparation/preparation.component.ts]: Yeah, it wasn't for `selectableTypes`, but this was just a placeholder type to pass to the dropdown where it is option, so I just removed the flag completely from this component.

**Review by mhuseinov — APPROVED** [review]
👍

**Pull request merged**